File: mobile_through_project_final/src/main/java/com/ssafy/cafe/model/dto/Product.java
package com.ssafy.cafe.model.dto;

import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class Product {
    private Integer id;
    private String name;
    private String type;
    private Integer price;
    private String img;
    private boolean isNew;
    private boolean isBest;
    private String content;
    private Integer kcal;
    
    @Builder
    public Product(Integer id, String name, String type, Integer price, String img, boolean isNew, boolean isBest, String content, Integer kcal) {
        super();
        this.id = id;
        this.name = name;
        this.type = type;
        this.price = price;
        this.img = img;
        this.isNew = isNew;
        this.isBest = isBest;
        this.content = content;
        this.kcal = kcal;
    }
    
    public Product(String name, String type, Integer price, String img, boolean isNew, boolean isBest) {
        this.name = name;
        this.type = type;
        this.price = price;
        this.img = img;
        this.isNew = isNew;
        this.isBest = isBest;
    }
}

File: app/src/main/java/com/ssafy/cobaltcoffee/dto/Product.kt
package com.ssafy.cobaltcoffee.dto

import com.google.gson.annotations.SerializedName

class Product(
    var id: Int,
    val name: String,
    val type: String,
    val price: Int,
    @SerializedName("img") val image: String,
    @SerializedName("new") val isNew: Boolean,
    @SerializedName("best") val isBest: Boolean
) : java.io.Serializable {
    constructor(): this(0, "", "", 0, "", false, false)
}